app/aimodels/bertopic/ai_services/basic_inference.py

```diff
@@ -404,10 +407,13 @@ def build_topic_model(self, topic_document_data: TopicDocumentData, num_topics,
         # split data, train, then infer. assumes documents, embeddings
         # sorted by timestamp previously (in train request)
         train_len = round(len(data_train) * train_percent)
-        data_test = data_train[train_len:]
-        data_train = data_train[:train_len-1]
+        if train_percent < 1.0:
+            data_test = data_train[train_len+1:]
+            data_train = data_train[:train_len]
```

Comment:
since the stop index in the slice is exclusive, it looks like we're leaving out the row at `train_len`?

![image](https://github.com/user-attachments/assets/055fbcfd-1dda-42dc-b3fe-d16a319a2665)

(should've caught this earlier since the previous code did something similar)